docs/subnets/customize-a-subnet.md

```diff
@@ -211,69 +211,100 @@ If `allowFeeRecipients` feature is enabled on the Subnet, but a validator doesn'
+
+`AllowList` adds `adminAddresses` and `enabledAddresses` fields to precompile contract configurations. For instance fee manager precompile contract configuration looks like this:
 
 ```json
 {
-  "config": {
-    "contractDeployerAllowListConfig": {
-      "blockTimestamp": 0,
-      "adminAddresses": ["0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"]
-    }
+  "feeManagerConfig": {
+    "blockTimestamp": 0,
+    "adminAddresses": [<list of addresses>],
+    "enabledAddresses": [<list of addresses>]
   }
 }
 ```
 
-In this example, `0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC` is named as the
-`Admin` of the `ContractDeployerAllowList`. This enables it to add other `Admins` or to add
-`Deployers`. Both `Admins` and `Deployers` can deploy contracts. To provide
-a great UX with factory contracts, the `tx.Origin` is checked for being a valid
-deployer instead of the caller of `CREATE`. This means that factory contracts will still be
-able to create new contracts as long as the sender of the original transaction is an allow
-listed deployer.
+`AllowList` configuration affects only the related precompile. For instance, the admin address in `feeManagerConfig` does not affect admin addresses in other activated precompiles.
 
-The `Stateful Precompile` powering the `ContractDeployerAllowList` adheres to the following Solidity interface at `0x0200000000000000000000000000000000000000` (you can load this interface and interact directly in Remix):
+The `AllowList` solidity interface is defined as follows:
 
 ```solidity
-// (c) 2022-2023, Ava Labs, Inc. All rights reserved.
-// See the file LICENSE for licensing terms.
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
 
-// SPDX-License-Identifier: MIT
+interface IAllowList {
+  // Set [addr] to have the admin role over the precompile
+  function setAdmin(address addr) external;
 
-pragma solidity >=0.8.0;
+  // Set [addr] to be enabled on the precompile contract.
+  function setEnabled(address addr) external;
 
-interface AllowListInterface {
-    // Set [addr] to have the admin role over the allow list
-    function setAdmin(address addr) external;
+  // Set [addr] to have no role the precompile contract.
+  function setNone(address addr) external;
 
-    // Set [addr] to be enabled on the allow list
-    function setEnabled(address addr) external;
+  // Read the status of [addr].
+  function readAllowList(address addr) external view returns (uint256 role);
+}
+```
+
+`readAllowList(addr)` will return a uint256 with a value of 0, 1, or 2, corresponding to the roles `None`, `Enabled`, and `Admin` respectively.
+
+_Note: `AllowList` is not an actual contract but just an interface. It's not callable by itself. This is used by other precompiles. Check other precompile sections to see how this works._
 
-    // Set [addr] to have no role over the allow list
-    function setNone(address addr) external;
+### Restricting Smart Contract Deployers
```

Comment:
```suggestion
#### Restricting Smart Contract Deployers
```
so that it is under the header `AllowList`?

Reply:
no. they should be equal.